packages/grant-explorer/src/features/round/ViewCartPage/SummaryContainer.tsx

```diff
@@ -275,13 +280,78 @@ export function SummaryContainer() {
     }
   }, [totalDonationAcrossChainsInUSDData]);
 
+  /* Matching estimates are calculated per-round */
+  const matchingEstimateParamsPerRound =
+    rounds?.map((round) => {
+      const projs = projects.filter((project) => project.roundId === round.id);
+      return {
+        roundId: getAddress(round.id ?? ""),
+        chainid: projs[0].chainId,
+        potentialVotes: projects.map((proj) => ({
+          amount: parseUnits(
+            proj.amount ?? "0",
+            getVotingTokenForChain(Number(proj.chainId) as ChainId).decimal ??
```

Comment:
`Number(chainIdAsString) as ChainId` would be a good candidate for a validating parser.

Reply:
Good idea - created a rudimentary one in the common package and replaced the casts in this file with it. More to be done.